core/services/directrequest/delegate.go

```diff
@@ -273,8 +273,19 @@ func (l *listener) handleOracleRequest(request *oracle_wrapper.OracleOracleReque
 		}
 		ctx, cancel := utils.CombinedContext(runCloserChannel, context.Background())
 		defer cancel()
-
-		_, _, err := l.pipelineRunner.ExecuteAndInsertFinishedRun(ctx, *l.job.PipelineSpec, pipeline.JSONSerializable{Val: meta, Null: false}, *logger, true)
+		run, trrs, err := l.pipelineRunner.ExecuteRun(ctx, *l.job.PipelineSpec, pipeline.JSONSerializable{Val: meta, Null: false}, *logger)
+		if ctx.Err() != nil {
+			return
+		} else if err != nil {
+			logger.Errorw("DirectRequest failed to create run", "err", err)
+		}
+		err = postgres.GormTransaction(ctx, l.db, func(tx *gorm.DB) error {
```

Comment:
It is probably cleaner to not use the parent context in DB operations, so we don't dirty-exit on application close.

Reply:
Does this context have a timeout deadline?

Reply:
Maybe we need a `postgres.GormTransactionWithDefaultContext` or something that does not take a ctx argument and always uses the default context

Reply:
gooood idea

Reply:
I think changing it to not cancel in flight jobs is a functional change we should separate that out from this PR